rows/plugins/xlsx.py

```diff
@@ -59,24 +63,26 @@ def import_from_xlsx(filename_or_fobj, sheet_name=None, sheet_index=0,
                      start_row=0, start_column=0, end_row=None,
                      end_column=None, *args, **kwargs):
     """Return a rows.Table created from imported XLSX file."""
-    
-    workbook = load_workbook(filename_or_fobj, data_only=True)
+
+    workbook = load_workbook(filename_or_fobj, data_only=True, read_only=True)
     if sheet_name is None:
         sheet_name = workbook.sheetnames[sheet_index]
     sheet = workbook[sheet_name]
 
     # openpyxl library reads rows and columns starting from 1 and ending on
     # sheet.max_row/max_col. rows uses another pattern: 0 to N - 1, so we need
     # to adjust the ranges accordingly
-    min_row, min_col = sheet.min_row - 1, sheet.min_column - 1
-    max_row, max_col = sheet.max_row - 1, sheet.max_column - 1
-    start_row = max(start_row, min_row)
-    end_row = min(end_row or max_row, max_row)
-    start_col = max(start_column, min_col)
-    end_col = min(end_column or max_col, max_col)
-    table_rows = [[_cell_to_python(sheet.cell(row=row_index, column=col_index))
-                   for col_index in range(start_col + 1, end_col + 2)]
-                  for row_index in range(start_row + 1, end_row + 2)]
+    column_offset = 1 if start_column or end_column else 0
+    row_offset = 1 if start_column or end_column else 0
+    sheet_rows = sheet.iter_rows(min_row=start_row, max_row=end_row,
+            min_col=start_column, max_col=end_column,
```

Comment:
This new code removes a very nice feature (which was not tested, but I've added tests on f41eec5629f2006d3e9e917f33db58780516cbeb): `sheet.min_row` and `sheet.max_row` will be automatically used to detect if the table having some padding (on the left and on the top), so if the table starts on cell C4, there will be no empty rows and cols on the returned `rows.Table`.